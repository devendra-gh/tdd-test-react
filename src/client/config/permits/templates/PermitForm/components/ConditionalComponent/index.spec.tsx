import React from 'react';
import { shallow } from 'enzyme';
import ConditionalFieldComponent from './index';

// eslint-disable-next-line no-console
console.log = () => {};
console.info = () => {};
console.warn = () => {};
console.error = () => {};

describe('Conditional Fields Component', () => {
  let props: any;
  beforeEach(() => {
    props = {
      i18n: jest.fn(),
      inputGroups: [
        {
          name: 'title.permitLocations',
          stateKey: 'permitLocations',
          rootcustomComponent: 'MultifieldForm',
          hasIncrementButton: false,
          incrementButtonLabel: 'Add Machines',
          fields: [
            {
              'aria-label': 'file upload',
              elementType: 'fileUpload',
              name: 'testName',
              accept: ['application/pdf'],
              label: 'test field',
              maxSize: 2e6,
              help: 'test help',
              conditionalBehaviour: () => true,
            },
          ],
        },
      ],
    };
  });

  it('should successfully render', () => {
    const component = shallow(<ConditionalFieldComponent {...props} />);
    expect(component).toMatchSnapshot();
  });
});
