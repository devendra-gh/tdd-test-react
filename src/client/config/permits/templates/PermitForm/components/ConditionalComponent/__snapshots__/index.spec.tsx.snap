// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Conditional Fields Component should successfully render 1`] = `null`;
